adapters/repos/db/inverted/searcher_doc_bitmap.go

```diff
@@ -17,9 +17,39 @@ import (
 
 	"github.com/weaviate/sroar"
 	"github.com/weaviate/weaviate/adapters/repos/db/lsmkv"
+	"github.com/weaviate/weaviate/adapters/repos/db/lsmkv/roaringset"
 	"github.com/weaviate/weaviate/entities/filters"
 )
 
+// MaxIDBuffer is the amount of bits greater than <maxDocID> to reduce
+// the amount of times BitmapFactory has to reallocate
+const MaxIDBuffer = uint64(1000)
+
+// BitmapFactory exists to prevent an expensive call to
+// NewBitmapPrefill each time NewInvertedBitmap is invoked
+type BitmapFactory struct {
```

Comment:
can be moved to `.../roaringset/helpers.go` where remaining BM related stuff is held (with `maxID...` renamed to `maxVal...` for consistency).